src/platform/Zephyr/SysHeapMalloc.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "SysHeapMalloc.h"
+
+#include <lib/support/CodeUtils.h>
+#include <system/SystemError.h>
+
+extern "C" {
+#include <init.h>
+#include <sys/mutex.h>
+#include <sys/sys_heap.h>
+}
+
+#include <cstdint>
+#include <cstring>
+
+// Construct name of the given function wrapped with the `--wrap=symbol` GCC option.
+#define WRAP(f) __wrap_##f
+
+using namespace chip;
+
+namespace {
+
+// Alignment of memory blocks returned by malloc.
+// Choose the value that guarantees that the returned memory blocks are castable to all built-in types.
+constexpr size_t kMallocAlignment = alignof(long long);
+
+uint8_t sHeapMemory[CONFIG_CHIP_MALLOC_SYS_HEAP_SIZE] alignas(kMallocAlignment);
+sys_heap sHeap;
+SYS_MUTEX_DEFINE(sLock);
+
+// RAII helper for synchronizing access to the common heap.
+class LockGuard
+{
+public:
+    LockGuard() : mStatus(sys_mutex_lock(&sLock, K_FOREVER)) {}
+    ~LockGuard();
+
+    bool Locked() const { return mStatus == 0; }
+    CHIP_ERROR Error() const { return System::MapErrorZephyr(mStatus); }
+
+private:
+    const int mStatus;
```

Comment:
kStatus?

Reply:
I think `kPrefix` is used for compile-time constants.

Reply:
Ok, I thought it is reserved for all constants, but maybe you are right.